var/lib/delphix-platform/ansible/10-delphix-platform/roles/delphix-platform/tasks/main.yml

```diff
@@ -234,3 +234,31 @@
     line: '{{ item.key }}="{{ item.value }}"'
   with_items:
     - { key: 'JAVA_HOME', value: '/usr/lib/jvm/oracle-java8-jdk-amd64' }
+
+- shell: dpkg-query -Wf '${Package}' 'delphix-platform-*' | sed 's/delphix-platform-//'
+  register: determine_platform
```

Comment:
We have precedence for setting variables via the `apply` script; e.g. see [here](https://github.com/delphix/delphix-platform/blob/master/var/lib/delphix-platform/ansible/apply#L38)

Would it be better to set the variable in `apply` rather than doing it here? If we do it that way, then any other roles (e.g. the app-gate) could also make use of it (not sure if it'll eventually need to or not).

i.e. we could do something like this:
```
PLATFORM=$(dpkg-query -Wf '${Package}' 'delphix-platform-*' | cut -d- -f 3 )

function apply_playbook() {
	if ! ansible-playbook -vvv \
...
                -e "platform=${PLATFORM}" \
```

and then it'd be used like what you're already doing:
```
when: platform is defined and platform == "azure"
```

perhaps we could even throw an error if `platform` isn't defined, since that probably means there's a bug rather than silently ignoring the issue:
```
- fail:
    msg: 'variable "platform" is not defined'
  when: platform is not defined
```
or
```
- assert:
    that:
      - platform is defined
```

If you'd rather go with what you have already implemented, that's OK with me too.

Reply:
That sounds reasonable to me, I'll do that.